src/languages/fsharp.js

```diff
@@ -1,20 +1,24 @@
 /*
 Language: F#
 Author: Jonas Follesø <jonas@follesoe.no>
-Contributors: Troy Kershaw <hello@troykershaw.com>, Henrik Feldt <henrik@haf.se>
+Contributors: Troy Kershaw <hello@troykershaw.com>, Henrik Feldt <henrik@haf.se>, Melvyn Laïly <melvyn.laily@gmail.com>
 Website: https://docs.microsoft.com/en-us/dotnet/fsharp/
 Category: functional
 */
 
 /** @type LanguageFn */
 export default function(hljs) {
+
+  const GENERICTYPESYMBOL = {
+    begin: /('|\^)[a-zA-Z0-9_]+/,
+    scope: 'symbol'
+  };
+  
   const TYPEPARAM = {
     begin: '<',
     end: '>',
     contains: [
-      hljs.inherit(hljs.TITLE_MODE, {
```

Comment:
Anytime you replace a built-in mode (`hljs.TITLE_MODE`) with a custom one first check it's relevance for example titles are just user-defined identifiers have `0` relevance to which language something is... so your new `GENERICTYPESYMBOL` rule should also have no relevance.

Plus there is already relevance here from the outer `<>` rule.  Is any amount of whitespace allowed here or could we make this rule tighter?

Reply:
> for example titles are just user-defined identifiers have 0 relevance to which language something is

Noted!

> Is any amount of whitespace allowed here or could we make this rule tighter?


Any amount of whitespace is allowed, as long as it's properly indented. The following is a valid construct:
```fsharp
type Blah<  
         'a 
         > = { x: 'a }
```

Reply:
I think you need a look-ahead regex on the `<` just to make sure it's followed by `whitespace '[char]`... otherwise this is set going to find tons of false positives. In fact this whole thing might be better as a multi-match (where you pass an array of regex and then pass scopes for each - see the docs) - if the type symbols always follow this pattern so closely.